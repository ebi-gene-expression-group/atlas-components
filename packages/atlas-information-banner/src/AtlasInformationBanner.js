import React from 'react'
import PropTypes from 'prop-types'
import ReactMarkdown from 'react-markdown'
import remarkGfm from 'remark-gfm'
import rehypeRaw from 'rehype-raw'

// We’re leveraging EBI Visual Framework (i.e. Foundation) to make the component functional. See this package’s
// html/index.html for a working example with all the necessary boilerplate.
// For vanilla HTML see version 1 of this component.
const AtlasInformationBanner = ({motd}) => {
  return (typeof motd === `string` && motd.trim().length > 0) ?
    (
      <div className={`callout`} data-closable={``}>
        <ReactMarkdown remarkPlugins={[remarkGfm]} rehypePlugins={[rehypeRaw]}>{motd}</ReactMarkdown>
        <button className={`close-button`} aria-label={`Dismiss information banner`} type={`button`} data-close={``}>
          <span aria-hidden={true}>&times;</span>
        </button>
      </div>
    ) :
    null
}

AtlasInformationBanner.propTypes = {
  motd: PropTypes.string.isRequired
}

export default AtlasInformationBanner
