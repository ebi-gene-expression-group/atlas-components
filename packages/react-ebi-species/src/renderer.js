"use strict";

//*------------------------------------------------------------------*

var React = require('react');
var ReactDOM = require('react-dom');

//*------------------------------------------------------------------*

var Icon = require('./SpeciesIcon.jsx');

//*------------------------------------------------------------------*

module.exports = function(mountNode,species,colourOverride, colourPerKingdom) {
    ReactDOM.render(
        React.createElement(Icon,{
          species: species,
          colourOverride: colourOverride,
          colourPerKingdom: colourPerKingdom
        }), mountNode
    );
};
