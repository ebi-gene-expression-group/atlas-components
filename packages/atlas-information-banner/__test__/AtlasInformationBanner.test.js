// If you get ReferenceError: regeneratorRuntime is not defined use this in your tests:
// import '@babel/polyfill'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'

import AtlasInformationBanner from '../src/AtlasInformationBanner'

describe(`AtlasInformationBanner`, () => {

	test(`disappears when close button is clicked`, () => {
		const wrapper = shallow(<AtlasInformationBanner/>)

		expect(wrapper.state('visibility')).toHaveLength(0);
		wrapper.find('a').last().simulate('click')
		expect(wrapper.state('visibility')).toBe('none');
	})

	test(`matches snapshot`, () => {
		const tree = renderer.create(<AtlasInformationBanner/>).toJSON()

		expect(tree).toMatchSnapshot()
	})
})
