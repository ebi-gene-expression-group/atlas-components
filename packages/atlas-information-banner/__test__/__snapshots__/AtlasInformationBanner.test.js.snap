// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtlasInformationBanner matches snapshot with a typical MOTD 1`] = `
<div
  className="callout"
  data-closable=""
  role="complementary"
>
  **Help us improve Single Cell Expression Atlas**

We are continuously developing the resource to deliver the best possible service for the community;
Please take two minutes to fill out our user survey and help us make Expression Atlas even better.

&lt;a target={_blank} href="https://www.surveymonkey.co.uk/r/SCEAsurvey22" class="button"&gt;Take survey&lt;/a&gt;
  <button
    aria-label="Dismiss information banner"
    className="close-button"
    data-close=""
    type="button"
  >
    <span
      aria-hidden={true}
    >
      Ã—
    </span>
  </button>
</div>
`;
